pyked/chemked.py

```diff
@@ -233,6 +241,238 @@ def get_dataframe(self, output_columns=None):
+            # initial composition is common
+            common.append('composition')
+            prop = etree.SubElement(common_properties, 'property')
+            prop.set('name', 'initial composition')
+
+            for species in composition:
+                component = etree.SubElement(prop, 'component')
+                species_link = etree.SubElement(component, 'speciesLink')
+                species_link.set('preferredKey', species['species-name'])
+                if species.get('InChI'):
+                    species_link.set('InChI', species['InChI'])
+
+                amount = etree.SubElement(component, 'amount')
+                amount.set('units', composition_type)
+                amount.text = str(species['amount'].magnitude)
+
+        # If multiple datapoints present, then find any common properties. If only
+        # one datapoint, then composition should be the only "common" property.
+        if len(self.datapoints) > 1:
+            for prop_name in datagroup_properties:
+                attribute = prop_name.replace(' ', '_')
+                quantity = getattr(self.datapoints[0], attribute)
+                if (quantity != None and
+                    all([quantity == getattr(dp, attribute) for dp in self.datapoints])
+                    ):
+                    common.append(prop_name)
+                    prop = etree.SubElement(common_properties, 'property')
+                    prop.set('description', '')
+                    prop.set('name', prop_name)
+                    prop.set('units', str(quantity.units))
+
+                    value = etree.SubElement(prop, 'value')
+                    value.text = str(quantity.magnitude)
+
+        # Ignition delay can't be common, unless only a single datapoint.
+
+        datagroup = etree.SubElement(root, 'dataGroup')
+        datagroup.set('id', 'dg1')
+        datagroup_link = etree.SubElement(datagroup, 'dataGroupLink')
+        datagroup_link.set('dataGroupID', '')
+        datagroup_link.set('dataPointID', '')
+
+        property_idx = {}
+        labels = {'temperature': 'T', 'pressure': 'P',
+                  'ignition delay': 'tau', 'pressure rise': 'dP/dt',
+                  }
+
+        for prop_name in datagroup_properties:
+            attribute = prop_name.replace(' ', '_')
+            if (prop_name not in common and
+                any([getattr(dp, attribute, None) for dp in self.datapoints])
+                ):
+                prop = etree.SubElement(datagroup, 'property')
+                prop.set('description', '')
+                prop.set('name', prop_name)
+                prop.set('units', str(getattr(self.datapoints[0], attribute).units))
+                idx = 'x{}'.format(len(property_idx) + 1)
+                property_idx[idx] = prop_name
+                prop.set('id', idx)
+                prop.set('label', labels[prop_name])
+
+        if 'composition' not in common:
+            for species in self.datapoints[0].composition:
```

Comment:
The composition of all the datapoints are not guaranteed to have the same species, so this has to be changed. You have a test for this, but it doesn't assert anything, so missed this case